hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -374,6 +395,191 @@ def finalize(self, settings, item):
+
+        ui_settings = {}
+
+        # Get the Publish Mode settings value from the UI combobox
+        publish_mode_combobox = widget.property("publish_mode_combobox")
+        if publish_mode_combobox:
+            mode_index = publish_mode_combobox.currentIndex()
+            if 0 <= mode_index < len(self.PUBLISH_MODE_OPTIONS):
+                self.PUBLISH_MODE_OPTIONS[mode_index]
+                ui_settings["Publish Mode"] = self.PUBLISH_MODE_OPTIONS[mode_index]
+            else:
+                self.logger.debug(f"Invalid Publish Mode index {mode_index}")
+
+        return ui_settings
+
+    def set_ui_settings(self, widget, settings, items=None):
+        """
+        This method is required to be defined in order for the custom UI to show up in the app.
+
+        Allows the custom UI to populate its fields with the settings from the
+        currently selected tasks.
+
+        The widget is the widget created and returned by
+        `create_settings_widget`.
+
+        A list of settings dictionaries are supplied representing the current
+        values of the settings for selected tasks. The settings dictionaries
+        correspond to the dictionaries returned by the settings property of the
+        hook.
+
+        Example::
+
+            settings = [
+            {
+                 "seeting_a": "/path/to/a/file"
+                 "setting_b": False
+            },
+            {
+                 "setting_a": "/path/to/a/file"
+                 "setting_b": False
+            }]
+
+        The default values for the settings will be the ones specified in the
+        environment file. Each task has its own copy of the settings.
+
+        When invoked with multiple settings dictionaries, it is the
+        responsibility of the custom UI to decide how to display the
+        information. If you do not wish to implement the editing of multiple
+        tasks at the same time, you can raise a ``NotImplementedError`` when
+        there is more than one item in the list and the publisher will inform
+        the user than only one task of that type can be edited at a time.
+
+        :param widget: The widget that was created by `create_settings_widget`.
+        :param settings: a list of dictionaries of settings for each selected
+            task.
+        :param items: A list of PublishItems the selected publish tasks are parented to.
+        """
+
+        if not settings:
+            return
+
+        if len(settings) > 1:
+            raise NotImplementedError
+
+        publish_mode_combobox = widget.property("publish_mode_combobox")
+        if not publish_mode_combobox:
+            self.logger.debug(
+                "Failed to retrieve Publish Mode combobox to set custom UI"
+            )
+            return
+
+        # Get the default setting for publish mode
+        default_value = (
+            self.settings.get("Publish Mode", {}).get("default")
+            or self.PUBLISH_MODE_DEFAULT
+        )
```

Comment:
I don't really understand this. It looks like there are 2 default values...?

Reply:
This is a fallback in case the setting "Publish Mode" was not defined. We do this in cases where it is not guaranteed the settings are defined. In this case though, it is redundant since this plugin itself defines the setting. I will remove the extra fallback here